src/main/java/sirius/kernel/nls/NLS.java

```diff
@@ -1273,59 +1273,97 @@ public static <V> V parseUserString(Class<V> clazz, String string) {
     }
 
     /**
-     * Converts a given time range in milliseconds to a human-readable format using the current language
+     * Converts a given time range to a human-readable format using the current language
      *
-     * @param duration       the duration in milliseconds
+     * @param duration       the duration
      * @param includeSeconds determines whether to include seconds or to ignore everything below minutes
      * @param includeMillis  determines whether to include milliseconds or to ignore everything below seconds
      * @return a string representation of the given duration in days, hours, minutes and,
      * if enabled, seconds and milliseconds
      */
-    public static String convertDuration(long duration, boolean includeSeconds, boolean includeMillis) {
+    public static String convertDuration(Duration duration, boolean includeSeconds, boolean includeMillis) {
         StringBuilder result = new StringBuilder();
-        if (duration > DAY) {
-            appendDurationValue(result, "NLS.days", duration / DAY);
-            duration = duration % DAY;
-        }
-        if (duration > HOUR) {
-            appendDurationValue(result, "NLS.hours", duration / HOUR);
-            duration = duration % HOUR;
-        }
-        if (duration > MINUTE || (!includeSeconds && duration > 0)) {
-            appendDurationValue(result, "NLS.minutes", duration / MINUTE);
-            duration = duration % MINUTE;
-        }
-        if (includeSeconds) {
-            if (duration > SECOND || (!includeMillis && duration > 0)) {
-                appendDurationValue(result, "NLS.seconds", duration / SECOND);
-                duration = duration % SECOND;
-            }
-            if (includeMillis && duration > 0) {
-                appendDurationValue(result, "NLS.milliseconds", duration);
-            }
+
+        long days = duration.toDaysPart();
+        if (days > 0) {
+            appendDurationValue(result, "NLS.days", days);
+        }
+        int hours = duration.toHoursPart();
+        if (hours > 0) {
+            appendDurationValue(result, "NLS.hours", hours);
+        }
+        int minutes = duration.toMinutesPart();
+        if (minutes > 0) {
+            appendDurationValue(result, "NLS.minutes", minutes);
+        }
+        int seconds = duration.toSecondsPart();
+        if (includeSeconds && seconds > 0 || includeMillis) {
+                appendDurationValue(result, "NLS.seconds", seconds);
+        }
+        int millis = duration.toMillisPart();
+        if (includeMillis && millis > 0) {
+            appendDurationValue(result, "NLS.milliseconds", millis);
         }
```

Comment:
Syntactically this is not entirely the same logic as before.
Before: includeMillis only was applied when includeSeconds AND includeMillis is set to true
After: includeMillis can be set to true while includeSeconds is disabled